test/user_test.dart

```diff
@@ -116,4 +116,332 @@ Future<void> main([List<String>? args]) async {
+
+    expect(fetched[1].id, second.id);
+    expect(fetched[1].isEnabled, true);
+
+    await user.apiKeys.enable(first.id);
+
+    final refetched = await user.apiKeys.fetchAll();
+    expect(refetched[0].id, first.id);
+    expect(refetched[0].isEnabled, true);
+
+    expect(refetched[1].id, second.id);
+    expect(refetched[1].isEnabled, true);
+  });
+
+  baasTest('User.apiKeys can login with generated key', (configuration) async {
+    final app = App(configuration);
+    final user = await getIntegrationUser(app);
+
+    final key = await user.apiKeys.create('my-key');
+    final credentials = Credentials.apiKey(key.value!);
+
+    final apiKeyUser = await app.logIn(credentials);
+    expect(apiKeyUser.provider, AuthProviderType.apiKey);
+    expect(apiKeyUser.id, user.id);
+    expect(apiKeyUser.refreshToken, isNot(user.refreshToken));
+  });
+
+  baasTest('User.apiKeys can login with reenabled key', (configuration) async {
+    final app = App(configuration);
+    final user = await getIntegrationUser(app);
+
+    final key = await user.apiKeys.create('my-key');
+
+    await user.apiKeys.disable(key.id);
+
+    final credentials = Credentials.apiKey(key.value!);
+
+    await expectLater(
+        () => app.logIn(credentials),
+        throwsA(isA<AppException>()
+            .having((e) => e.message, 'message', contains('invalid API key'))
+            .having((e) => e.statusCode, 'statusCode', 401)
+            .having((e) => e.linkToServerLogs, 'linkToServerLogs', contains('logs?co_id='))));
+
+    await user.apiKeys.enable(key.id);
+
+    final apiKeyUser = await app.logIn(credentials);
+    expect(apiKeyUser.provider, AuthProviderType.apiKey);
+    expect(apiKeyUser.id, user.id);
+    expect(apiKeyUser.refreshToken, isNot(user.refreshToken));
+  });
+
+  baasTest("User.apiKeys can't login with deleted key", (configuration) async {
+    final app = App(configuration);
+    final user = await getIntegrationUser(app);
+
+    final key = await user.apiKeys.create('my-key');
+
+    await user.apiKeys.delete(key.id);
+
+    final credentials = Credentials.apiKey(key.value!);
+
+    await expectLater(
+        () => app.logIn(credentials),
+        throwsA(isA<AppException>()
+            .having((e) => e.message, 'message', contains('invalid API key'))
+            .having((e) => e.statusCode, 'statusCode', 401)
+            .having((e) => e.linkToServerLogs, 'linkToServerLogs', contains('logs?co_id='))));
+  });
+
+  baasTest("Credentials.serverApiKey can login user", (configuration) async {
+    final app = App(configuration);
+
+    final apiKey = await createServerApiKey(app, ObjectId().toString());
+    final credentials = Credentials.serverApiKey(apiKey);
+
+    final apiKeyUser = await app.logIn(credentials);
+
+    // This is due to the way Core stores the user provider: https://github.com/realm/realm-core/blob/7b087008a19773f1bb92324d410f9e70db17fca9/src/realm/object-store/sync/app_credentials.cpp#L33-L34
```

Comment:
I think it is correct to create an issue for Core.

Reply:
I asked on #realm-core-team - I was hoping to get a response, but can file a ticket as well.

Reply:
Filed https://github.com/realm/realm-core/issues/5914.